edge-apps/clock/src/static/js/main.js

```diff
@@ -0,0 +1,76 @@
+// eslint-disable-next-line no-unused-vars
+function initApp () {
+  let clockTimer
+  let timeFormat = 'h12'
+
+  const locale = navigator?.languages?.length
+    ? navigator.languages[0]
+    : navigator.language
+
+  const getDefaultTimePreference = () => {
+    const url = new URL(window.location)
+    const preference = url.searchParams.get('24h')
+
+    if (typeof preference === 'string') {
+      return preference === '0' ? 'h12' : 'h24'
+    }
+  }
+
+  const setTimeFormat = () => {
+    const preference = getDefaultTimePreference()
+    if (preference) {
+      timeFormat = preference
+    } else {
+      timeFormat = Intl.DateTimeFormat(locale, { hour: 'numeric' }).resolvedOptions().hourCycle || 'h12'
+    }
+  }
+
+  const convert24to12format = (hrs) => hrs % 12 || 12
+
+  const padTime = (time) => String(time).padStart(2, '0')
+
+  const formatTimeByLocale = (hrs, mins) => {
```

Comment:
Can we just use (momentjs)[https://momentjs.com/) here instead of trying to hack this together ourselves?